balena/models/device.py

```diff
@@ -1033,7 +1033,41 @@ def set_to_release(self, uuid, commit_id):
         """
 
         device_id = self.get(uuid)['id']
-        release_id = self.release._Release__get_by_option('commit', commit_id)[0]['id'] if commit_id else None
+        release_id = self.release._Release__get_by_option('commit', commit_id, status='success')[0]['id'] if commit_id else None
+
+        params = {
```

Comment:
It seems that we do have a bit of repetition in here, how about making `set_to_release` to just fetch the release id and then call `set_to_release_by_id` for all the rest?

Reply:
Makes sense, updated.